go.mod

```diff
@@ -7,14 +7,19 @@ toolchain go1.23.4
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/go-logr/logr v1.4.2
+	github.com/manifestival/controller-runtime-client v0.4.0
```

Comment:
What's the reason for bumping these deps? Generally we probably shouldn't do it unless it's necessary, or part of release chores, and if we need to, it's best to put it in a separate commit to keep it contained and easier to review and view past history..